OpenCV using C's math instead of C++'s cmath making build error

##### System information (version)
<!-- Example
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2015
-->

- OpenCV => 3.2.0
- Operating System / Platform => Debian Stretch 64 Bit
- Compiler => GCC 6

##### Detailed description

I was building OpenCV when I got this error - `farneback_optical_flow.cpp:29:22: error: call of overloaded ‘abs(double&)’ is ambiguous`.

The fix was to add this `#include <cmath>` to the beginning of the file `farneback_optical_flow.cpp`.

##### Steps to reproduce

Build OpenCV with Examples and Tests.

--- a/samples/gpu/farneback_optical_flow.cpp
+++ b/samples/gpu/farneback_optical_flow.cpp
@@ -1,6 +1,7 @@
 #include <iostream>
 #include <vector>
 #include <sstream>
+#include <cmath>
 
 #include "opencv2/core.hpp"
 #include "opencv2/core/utility.hpp"

